eng/pipelines/templates/stages/archetype-ios-release.yml

```diff
@@ -66,17 +66,32 @@ stages:
 
           - deployment: PublishPackageToSwiftPackageManagerMirrors
             displayName: "Publish to SPM Mirrors"
-            condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'))
+            condition: and(succeeded(), ne(variables['Skip.PublishPackage'], 'true'), ne('${{artifact.skipSwiftPackageManager}}', true))
             environment: cocoapodstrunk
             dependsOn: TagRepository
             variables:
               - template: ../variables/globals.yml
             pool:
-              name: Azure Pipelines
-              vmImage: $(OSVmImage)
+              name: azsdk-pool-mms-ubuntu-1804-general
+              vmImage: MMSUbuntu18.04
             strategy:
               runOnce:
                 deploy:
                   steps:
-                    - script: |
-                        echo "Pretend we are publishing to Swift Package Manager mirrors"
+                    - checkout: self
+                    - checkout: SwiftPM-${{artifact.name}}
+                      persistCredentials: true
+                    - pwsh: |
+                        $podspec = Get-Content -Raw -Path $(Pipeline.Workspace)/s/azure-sdk-for-ios/sdk/${{parameters.ServiceDirectory}}/${{artifact.name}}/${{artifact.name}}.podspec.json | ConvertFrom-Json
+                        $version = $podspec.version
+                        Push-Location -Path $(Pipeline.Workspace)/s/SwiftPM-${{artifact.name}}
+                        git checkout --orphan run-$(Build.BuildID)-temp-branch
+                        git reset --hard
+                        Copy-Item -Path $(Pipeline.Workspace)/s/azure-sdk-for-ios/sdk/${{parameters.ServiceDirectory}}/${{artifact.name}}/* -Destination $(Pipeline.Workspace)/s/SwiftPM-${{artifact.name}} -Recurse
+                        git add .
```

Comment:
We should try  to use our existing templates https://github.com/Azure/azure-sdk-tools/blob/master/eng/common/pipelines/templates/steps/git-push-changes.yml.

Reply:
I think that these are very different scenarios. In this case I'm pushing the tag, not the branch, I'm not sure what this script will provide to this code, but adding my scenario to the script would probably just introduce a lot of unnecessary complexity.

Reply:
Fair if you are only pushing the tag then I agree we probably don't need to add that complexity to that script. If we are simply pushing a branch and then tagging separately then we could consider using the common stuff but I agree this is a different scenario.

Reply:
Yeah the branch never goes to the server, I'm only using it as a way of creating an orphan parent commit.